src/Symfony/Component/DependencyInjection/Attribute/Autowire.php

```diff
@@ -28,17 +30,21 @@ class Autowire
     /**
      * Use only ONE of the following.
      *
-     * @param string|null $value      Parameter value (ie "%kernel.project_dir%/some/path")
-     * @param string|null $service    Service ID (ie "some.service")
-     * @param string|null $expression Expression (ie 'service("some.service").someMethod()')
+     * @param string|array|null $value      Parameter value (ie "%kernel.project_dir%/some/path")
+     * @param string|null       $service    Service ID (ie "some.service")
+     * @param string|null       $expression Expression (ie 'service("some.service").someMethod()')
+     * @param string|null       $env        Environment variable name (ie 'SOME_ENV_VARIABLE')
+     * @param string|null       $param      Parameter name (ie 'some.parameter.name')
      */
     public function __construct(
         string|array $value = null,
         string $service = null,
         string $expression = null,
+        string $env = null,
+        string $param = null,
     ) {
-        if (!($service xor $expression xor null !== $value)) {
-            throw new LogicException('#[Autowire] attribute must declare exactly one of $service, $expression, or $value.');
+        if (!($service xor $expression xor $env xor $param xor null !== $value)) {
```

Comment:
`if (!(null !== $service xor null !== $expression xor null !== $env xor null !== $param xor null !== $value)) {`

Reply:
100% yes. Now I understand what this condition means :D

Done.